### ISSUE TITLE
Sozi changes one transition upon loading unmodified svg file...

### ISSUE THREAD

**Issue by lopippo** [sozi-projects/Sozi#573]
First off, thank you for this very enjoyable software ! 

### Summary of your problem or feature request

First case
=======

I craft a presentation that works fine (inside Sozi, the transitions are perfect). I save that presentation (no change to the Inkscape svg file) from Sozi ("Save the presentation" menu). I check the obtained html file, the transitions work perfectly.

I close all the programs. I now load the svg file in Sozi. As an unexpected result, the html and json files are changed. One of the transitions does not work anymore as it did before (that transition fails in the same manner inside Sozi and in the changed html file). 

Second case
=========

Same starting point as above, but instead of closing Sozi and then restarting it with the Inkscape svg file, I reload the Inkscape svg file (that did not change in any way). As an unexpected result, the html and json files are changed. One of the transitions does not work anymore as it did before (that transition fails in the same manner inside Sozi and in the changed html file). 

Question 1
========

Apart from the fact that one of the transitions fails after having worked flawlessly, why does the program write to files even when configured not to do so automatically? 

Question 2
========

### Version of Sozi and other relevant software

* Which version of Sozi are you using? sozi-21.11.14-1636892493-linux-x64
* Which drawing software did you use? (examples: Inkscape 0.48, LibreOffice Draw 4.2) Inscape 1.0.2 (e86c870879, 2021-01-15)
* On which platform? (examples: Windows 7 64-bit, Ubuntu Linux 14.04 32-bit, OS X El Capitan 64-bit) Debian GNU/Linux bullseye amd64
* If your problem happens when playing a presentation, which browsers have you tried? (examples: Firefox 46, Chrome 52)

Thank you for listening :-)

Filippo

**Comment by aumouvantsillage** [comment]
Thanks for reporting the issue.
Could you send me an example (SVG + JSON) that could help me reproduce the problem?

> Apart from the fact that one of the transitions fails after having worked flawlessly, why does the program write to files even when configured not to do so automatically?

Sozi always saves the presentation when loading the SVG, even if it is configured for manual saving.
I agree that it should be changed.

**Comment by aumouvantsillage** [comment]
This issue could be related to #575

**Comment by lopippo** [comment]
Greetings,

thank you for your response. I have reduced the presentation, by removing a number of frames. The frames left present the problem described above.

To reproduce, a little context:

I would like (and succeed in getting that to work) that the contents of the frames 6 and 7 be perfectly overlapped so that the text in orange color be disappearing/appearing exactly at the same place (no translation, just fading-out for exemple-ldtfm and fading-in for ldtfm-masse-mono.

If, in both frames, I set the rect-exemple-ldtfm and rect-ldtfm-masse-mono outline element Ids, then I achieve exactly that. In the frame number 6 (exemple-ldtfm), rect-ldtfm-masse-mono is set to opacity 0 and rect-exemple-ldtfm is set to opacity 1. Then, in frame number 7 (ldtfm-masse-mono), the exact reverse is done. 

When playing the presentation in this setup, inside Sozi, the animation is perfect. Now, when I reload the svg file that animation is lost.

The archive I provide here is in the exact configuration where it works (SVG file saved, presentation saved, the HTML file performs fine). I have added the HTML file so that you can play the working animation.

Please, find the archive at this place:

https://filesender.renater.fr/?s=download&token=01c5d789-53de-4561-9f26-cc14bf42e6c6

This archive is available for 30 days (max availability duration). If necessary, just ask and I'll repost it :-)

Yours, most sincerely,
Filippo

**Comment by aumouvantsillage** [comment]
There were two bugs in my code that were related to your problem.

One was triggered by the unexpected format of the SVG document (I assume that some parts were created outside of Inkscape).

For the other bug, it seems that Sozi could not reliably compute the position of a layer when it was completely transparent.
This is weird because in my own presentations, I use this technique a lot.

I will publish a new release in a few days. Hopefully, it will make Sozi more predictable. After you try the new version, please tell me if you still have this kind of issues.

**Comment by lopippo** [comment]
Remarkable ! Thank you so much.

By the way, I think any SVG in the document was created with Inkscape. However, the problem might be related to my heavily using TexText [an extension that allows crafting LaTeX-based pseudo-objects in Inkscape (not really textual objects, but editable any time anyway)]. Does this make sense to you ?

Of course shall I keep you posted after having tested the new version.

Yours, most sincerely,
Filippo

**Comment by aumouvantsillage** [comment]
> However, the problem might be related to my heavily using TexText [...] Does this make sense to you ?

Yes. As I understand it, the extension runs LaTeX to compile your text elements to SVG, and the result is injected into your main document in Inkscape.
The generated SVG is correct and we should expect Sozi to handle it correctly.

**Comment by aumouvantsillage** [comment]
You can try [Sozi 21.12](https://github.com/sozi-projects/Sozi/releases/tag/v21.12). It contains a fix that addresses this issue.
I cannot promise that it will automatically fix your existing presentations, but I hope it will prevent similar problems when you modify them or create new ones.

**Comment by aumouvantsillage** [comment]
I'm closing this issue, but feel free to reopen it if you think the problem has not been fully fixed.

**Comment by lopippo** [comment]
Greetings, 

thank you for the new version that I have tested right away.
I could not succeed in saving the presentation. I started afresh, by deleting the json and html files. I only had my svg file available. I could load it fine in Sozi. I created my first frame fine.

Upon trying to save the presentation manually, a json file appeaed but no html file was generated.
I thus tried the "save automatically when Sozi looses focus", but that did not help.

What a I missing ?

Sincerely,
Filippo
